src/server/pfs/server/val_server.go

```diff
@@ -0,0 +1,101 @@
+package server
+
+import (
+	"github.com/pachyderm/pachyderm/src/client"
+	"github.com/pachyderm/pachyderm/src/client/auth"
+	"github.com/pachyderm/pachyderm/src/client/pfs"
+	"github.com/pachyderm/pachyderm/src/client/pkg/errors"
+	"github.com/pachyderm/pachyderm/src/client/pkg/grpcutil"
+	"github.com/pachyderm/pachyderm/src/server/pkg/serviceenv"
+	"golang.org/x/net/context"
+)
+
+var _ APIServer = &valAPIServer{}
+
+type valAPIServer struct {
```

Comment:
I think this should be `validateAPIServer` or `validatedAPIServer`, then the constructor should be named appropriately (`new<full abstraction name>`).